### PR TITLE
[4.0] Extend JController from the Framework

### PR DESCRIPTION
**Pull request by mbabker**
### Summary of Changes

The web application routers were removed in the 4.0 branch in favor of using the Framework's Router package.  1.x of this package requires a `Joomla\Controller\ControllerInterface` implementation, of which it is impossible to do with existing CMS code.  This PR therefore makes the "no MVC" controller base class and interface compatible with the Framework's Controller package.

### Testing Instructions

Code review

### Documentation Changes Required

The `$app` and `$input` properties will now be private, requiring the use of the appropriate getters to retrieve this objects in subclasses.

If merged this will be added to https://docs.joomla.org/Potential_backward_compatibility_issues_in_Joomla_4

### CODE REVIEW

**Comment by wilsonge** [comment]
We need to test com_config to merge this - and just make sure that all still works as expected as that all runs on new mvc

**Comment by mbabker** [comment]
Rebased

File: libraries/joomla/controller/controller.php [hunk 1]
```diff
@@ -9,43 +9,14 @@
...
 
 /**
  * Joomla Platform Controller Interface
  *
- * @since  12.1
+ * @since       12.1
+ * @deprecated  5.0  Implement Joomla\Controller\ControllerInterface instead
  */
-interface JController extends Serializable
+interface JController extends ControllerInterface
```
Comment (wilsonge) [inline libraries/joomla/controller/controller.php]: I think we should probably leave `getInput` and `getApplication` in this interface. If people want the higher level interface they can move to the framework one

**Comment by mbabker** [comment]
Rebased and updated.

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**